plugins/org.eclipse.elk.core.service/src/org/eclipse/elk/core/service/ElkServicePlugin.java

```diff
@@ -53,18 +78,30 @@ public void stop(final BundleContext context) throws Exception {
         for (LayoutAlgorithmData algoData : layoutDataService.getAlgorithmData()) {
             algoData.getInstancePool().clear();
         }
+        if (executorService != null) {
+            executorService.shutdown();
+            executorService = null;
+        }
         
         plugin = null;
         super.stop(context);
     }
-
+    
     /**
-     * Returns the shared instance.
-     * 
-     * @return the shared instance
+     * Return the executor service to use for automatic layout operations.
      */
-    public static ElkServicePlugin getDefault() {
-        return plugin;
+    public synchronized ExecutorService getExecutorService() {
+        if (executorService == null && plugin != null) {
+            executorService = Executors.newCachedThreadPool();
+        }
+        return executorService;
+    }
+    
+    /**
+     * Return the map of currently running layout operations.
+     */
+    public Multimap<Pair<IWorkbenchPart, Object>, MonitoredOperation> getRunningOperations() {
```

Comment:
The synchronization is done in DiagramLayoutEngine, but we could move it to the plugin to avoid making the multimap public. Or we could move back the multimap into DiagramLayoutEngine and make it static.